action/action.php

```diff
@@ -107,4 +108,54 @@ public function handleCache(Doku_Event $event)
             http_status(500);
         }
     }
+
+    /**
+     * PNG download available via link created in JS (only if PNG caching is enabled)
+     *
+     * @param Doku_Event $event
+     * @return void
+     */
+    public function handlePNGDownload(Doku_Event $event)
+    {
+        if ($event->data !== 'plugin_diagrams_pngdownload') return;
+        $event->preventDefault();
+        $event->stopPropagation();
+
+        global $INPUT;
+        global $conf;
+
+        $cacheName = $INPUT->str('pngcache');
+        $media = cleanID($INPUT->str('media'));
+        $id = cleanID($INPUT->str('id'));
+
+        // check ACLs to original file or page
+        if (
+            ($id && auth_quickaclcheck($id) < AUTH_READ) ||
+            ($media && auth_quickaclcheck($media) < AUTH_READ)
+        ) {
```

Comment:
Hmm... I wonder... Would this allow me to get a diagram that I don't have access to if I know the cache ID? I could simply provide the $id or $media I do have access to, together with the cacheID I want and this would let me have it, wouldn't it? I'm not sure that's a realistic scenario...

Reply:
Since cacheID is generated from file content, it ts rather impossible to guess it.